src/adagrams.js

```diff
@@ -1,15 +1,138 @@
+const letterDistribution = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+let userLetters = [];
+const allLetters = [];
 export const drawLetters = () => {
   // Implement this method for wave 1
+   for (const letter in letterDistribution) {
+    for (let i = 0; i < letterDistribution[letter]; i++) {
+      allLetters.push(letter);
+    }
+  }
+
+  while (userLetters.length < 10) {
+    const randomLetterIndex = Math.floor(Math.random() * allLetters.length);
+    const randomLetter = allLetters[randomLetterIndex];
+
+    userLetters.push(randomLetter);
+    allLetters.splice(randomLetterIndex, 1);
+  }
```

Comment:
Rather than creating two separate scopes, you could have created a for loop that ran 10 times and combined this while loop and `for/in` into one. With a for loop you could have used i as an index and avoided the additional loop. Remember we should keep the algorithmic complexity in mind.

Remember to keep in mind where certain built-in function calls might be hiding a loop, and try to avoid using those within an outer loop (things like splice or includes). Even though the small size of the data in this means the absolute performance is still fast, we should keep the algorithmic complexity in mind. Imagine what would happen if we relaxed the data restrictions to use larger tile sets or hands.

Also be careful with indentation here 💯